apps/microtvm/stm32/Src/Validation/pb_decode.c

```diff
@@ -0,0 +1,1528 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file pb_decode.c
+ * \brief decode a protobuf using minimal resources.
+ *        2011 Petteri Aimonen <jpa@kapsi.fi>
+ */
```

Comment:
discussed with @tqchen and it's ok to check these in, but we do want to keep all non-apache-licensed files in 3rdparty/. Sorry this restriction isn't written up yet--we will make an effort to add it to the docs soon.

Reply:
Clear. Will it be OK to create an apps-stm32 sub-directory inside the 3rdparty directory that will contain the 5 files that we are talking about ? Checking in the nanopb looks too heavy for the use we make, additionally I would not know where to put the 2 files generated via the nanopb for our target.

Reply:
yeah that's fine. feel free to check-in to `3rdparty/stm32-runtime` and symlink as necessary.

Reply:
it looks like you've added the ASF header to these files in 3rdparty/stm32-runtime. i don't think we can do that, though, because they are not re-licensed under the Apache license. Only the original author can do that. Can you remove the headers from those files? i don't think you will need to, but you may need to tweak `tests/lint/rat-excludes` to make lint pass.